File: Sample/iOS_OSX/Sample-iOS/ModuleViewControllers/EncodingModuleViewController.m
//
//  EncodingModuleViewController.m
//  Sample
//
//  Created by 冯鸿杰 on 2018/8/21.
//  Copyright © 2018年 vimfung. All rights reserved.
//

#import "EncodingModuleViewController.h"
#import "LuaScriptCore.h"

@interface EncodingModuleViewController ()

@property (nonatomic, strong) LSCContext *context;

@end

@implementation EncodingModuleViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.context = [[LSCContext alloc] init];
    [self.context onException:^(NSString *message) {
        
        NSLog(@"lsc exception = %@", message);
        
    }];
    
    [self.context evalScriptFromFile:@"Encoding-Sample.lua"];
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    switch (indexPath.row)
    {
        case 0:
            //Url Encode
            [self.context evalScriptFromString:@"Encoding_Sample_urlEncode()"];
            break;
        case 1:
            //Url Decode
            [self.context evalScriptFromString:@"Encoding_Sample_urlDecode()"];
            break;
        case 2:
            //Base64 Encode
            [self.context evalScriptFromString:@"Encoding_Sample_base64Encode()"];
            break;
        case 3:
            //Base64 Decode
            [self.context evalScriptFromString:@"Encoding_Sample_base64Decode()"];
            break;
        case 4:
            //JSON Encode
            [self.context evalScriptFromString:@"Encoding_Sample_jsonEndode()"];
            break;
        case 5:
            //JSON Decode
            [self.context evalScriptFromString:@"Encoding_Sample_jsonDecode()"];
            break;
        case 6:
            //Hex Encode
            [self.context evalScriptFromString:@"Encoding_Sample_hexEncode()"];
            break;
        case 7:
            //Hex Decode
            [self.context evalScriptFromString:@"Encoding_Sample_hexDecode()"];
            break;
        default:
            break;
    }
}

@end

File: Sample/iOS_OSX/Sample-iOS-Swift/ModuleViewControllers/EncodingModuleViewController.swift
//
//  EncodingModuleViewController.swift
//  Sample-iOS-Swift
//
//  Created by 冯鸿杰 on 2018/8/21.
//  Copyright © 2018年 vimfung. All rights reserved.
//

import UIKit
import LuaScriptCore_iOS_Swift

class EncodingModuleViewController: UITableViewController {

    let context : LuaContext = LuaContext();
    
    override func viewDidLoad() {
        
        context.onException { (msg) in
            print("lua exception = \(msg!)");
        }
        
        _ = context.evalScript(filePath: "Encoding-Sample.lua");
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        switch indexPath.row
        {
        case 0:
           _ = context.evalScript(script: "Encoding_Sample_urlEncode()");
        case 1:
           _ = context.evalScript(script: "Encoding_Sample_urlDecode()");
        default:
            break;
        }
    }
    
}
